packages/test/test-gc-sweep-tests/src/dataObjectWithChildDataObject.ts

```diff
@@ -0,0 +1,71 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { DataObjectFactory } from "@fluidframework/aqueduct";
+import { assert } from "@fluidframework/common-utils";
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { DataObjectWithCounter, dataObjectWithCounterFactory } from "./dataObjectWithCounter";
+import { allFactories } from "./testDataObjects";
+
+const childKey = "childKey";
+export class DataObjectWithChildDataObject extends DataObjectWithCounter {
+    private unrefTimeStampMs?: number;
+    private oldChild?: DataObjectWithCounter;
+    private child?: DataObjectWithCounter;
+
+    public static get type(): string {
+        return "DataObjectWithChildDataObject";
+    }
+
+    protected async initializingFirstTime(props?: any): Promise<void> {
+        await super.initializingFirstTime(props);
+        await this.referenceNewChild();
+    }
+
+    protected async hasInitialized(): Promise<void> {
+        await super.hasInitialized();
+        const childHandle = this.root.get<IFluidHandle<DataObjectWithCounter>>(childKey);
+        this.child = await childHandle?.get();
```

Comment:
You should assert that the childHandle exists.

Reply:
With our new implementation this problem should go away as on hasInitialized we would be creating our child dataobject